### DISCUSSION TITLE
How to actually include modules for custom grammars?

### DISCUSSION THREAD

**Discussion by madskjeldgaard** [nvim-treesitter/nvim-treesitter#885] [Q&A] [answered]
Hello
Thanks for all your had work on this project. I love it and use it as a daily driver :)

I am writing a grammar for a programming language and it is now at the stage where I can start to use it for highlighting in nvim. 

I have run into some issues though when following the custom grammar installation steps on here. I managed to install it, and it shows up on  `TSInstallInfo` as installed, so all seems good, but the thing I don't fully understand from the instructions is how to activate modules for a custom grammar. When running `TSModuleInfo` I can see that my custom grammar has deactivated modules for all three columns, even though it includes for example a highlights.scm file for syntax highlighting.

Looking forward to hearing from you  ! 

best

**Comment by madskjeldgaard** [comment]
Oh and I should of course include my config for treesitter:
```lua
-- tree-sitter-supercollider
	local parser_config = require "nvim-treesitter.parsers".get_parser_configs()
	parser_config.supercollider = {
		install_info = {
			url = "~/code/tree-sitter-supercollider",
			files = {"src/parser.c"}
		},
		filetype = "supercollider", -- if filetype does not agrees with parser name
		used_by = {"scd", "sc"} -- additional filetypes that use this parser
	}


	require'nvim-treesitter.configs'.setup {
		indent = {
			enable = true
		},
		ensure_installed = "maintained", -- one of "all", "maintained" (parsers with maintainers), or a list of languages
		highlight = {
			enable = true,              -- false will disable the whole extension
			-- disable = { "c", "rust", "lua", "cpp", "html" , "python", "bash"},  -- list of language that will be disabled
		},
		incremental_selection = {
			enable = true,
			keymaps = {
				init_selection = "gnn",
				node_incremental = "grn",
				scope_incremental = "grc",
				node_decremental = "grm",
			},
		},
	}

```

**Comment by vigoux** [comment]
I am not sure to understand what you are expecting and does not work here...
First, are you sure that the filetype for your file is correctly set ? You might be missing things like `ftdetect` and so on.

**Comment by madskjeldgaard** [comment]
Hi 

I just don't really understand how to activate modules for my language after installation. 
Inside a file of my filetype I can run  `lua require'nvim-treesitter'.parsers.has_parser()` which returns true so I think the filetype detection part is fine? 

My expectation is that I see the tree-sitter syntax highlighting in action at least :) 

`set filetype` returns `supercollider` which is what my parser is expecting (see config above) I think.

**Reply by vigoux** [reply]
Okay, I seem to be able to reproduce, so that looks like a bug to me.

**Reply by vigoux** [reply]
Though, i can't reproduce when opening a file from `:e` or the command line, are you using telescope or something ?

**Comment by madskjeldgaard** [comment]
No, I don't use telescope, but I should add that I am on nvim bleeding edge 0.5 build :) 
I use fzf/skim for file opening. I can test here as well. What symptom of success exactly with `:e` is it you are seeing? Then I'll look for that here as well. Thanks!

**Reply by vigoux** [reply]
The symptoms are that the file is successfully highlighted 😉

**Reply by madskjeldgaard** [reply]
ah! hahhaha! Okay then yeah! 
Well I don't get any highlighting at all no matter how I open. And just for reference as well, this is the (work in progress) grammar I am testing with: https://github.com/madskjeldgaard/tree-sitter-supercollider

**Reply by vigoux** [reply]
Could you paste the output of `:checkhealth nvim-treesitter` specifically the part for your language ?

**Reply by madskjeldgaard** [reply]
aha, it doesn't seem to be able to find the queries
```
## supercollider parser healthcheck
  - OK: supercollider parser found.
  - WARNING: No `highlights.scm` query found for supercollider
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `locals.scm` query found for supercollider
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `folds.scm` query found for supercollider
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `indents.scm` query found for supercollider
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter


```

**Reply by vigoux** [reply]
Where did you put your query ?

**Reply by madskjeldgaard** [reply]
The query is in a `queries` subfolder in the url I supplied in the parser config `url = "~/code/tree-sitter-supercollider",`

**Reply by madskjeldgaard** [reply]
Does that get copied when you install a parser like this?

**Reply by vigoux** [reply]
Nope, you need to have that in your runtime path, so something like `.config/nvim/queries`

**Reply by madskjeldgaard** [reply]
ahhh

**Reply by madskjeldgaard** [reply]
Yes that definately seems to solve it

**Comment by madskjeldgaard** [comment, marked as answer]
For future reference, I solved this by symbolically linking my custom parser's queries to my nvim runtime like this:
```
ln -s /home/mads/code/tree-sitter-supercollider/queries /home/mads/.config/nvim/queries/supercollider
```
